addons/account_edi_proxy_client/models/account_edi_proxy_user.py

```diff
@@ -45,28 +42,63 @@ class AccountEdiProxyClientUser(models.Model):
     company_id = fields.Many2one('res.company', string='Company', required=True,
         default=lambda self: self.env.company)
-    edi_format_id = fields.Many2one('account.edi.format', required=True)
-    edi_format_code = fields.Char(related='edi_format_id.code', readonly=True)
-    edi_identification = fields.Char(required=True, help="The unique id that identifies this user for on the edi format, typically the vat")
+    edi_identification = fields.Char(required=True, help="The unique id that identifies this user, typically the vat")
     private_key = fields.Binary(required=True, attachment=False, groups="base.group_system", help="The key to encrypt all the user's data")
     private_key_filename = fields.Char(compute='_compute_private_key_filename')
     refresh_token = fields.Char(groups="base.group_system")
+    proxy_type = fields.Selection(selection=[], string='Proxy type')
+    edi_mode = fields.Selection(
+        selection=[
+            ('prod', 'Production mode'),
+            ('test', 'Test mode'),
+            ('demo', 'Demo mode'),
+        ],
+        string='EDI operating mode',
+        compute='_compute_edi_mode', readonly=False, store=True,
+    )
 
     _sql_constraints = [
         ('unique_id_client', 'unique(id_client)', 'This id_client is already used on another user.'),
-        ('unique_edi_identification_per_for', 'unique(edi_identification, edi_format_id)', 'This edi identification is already assigned to a user'),
+        ('unique_edi_identification', 'unique(edi_identification, proxy_type, edi_mode)', 'This edi identification is already assigned to a user'),
     ]
 
     def _compute_private_key_filename(self):
         for record in self:
             record.private_key_filename = f'{record.id_client}_{record.edi_identification}.key'
 
-    def _get_demo_state(self):
-        demo_state = self.env['ir.config_parameter'].sudo().get_param('account_edi_proxy_client.demo', False)
-        return 'prod' if demo_state in ['prod', False] else 'test' if demo_state == 'test' else 'demo'
+    def _compute_edi_mode(self):
+        for user in self:
+            user.edi_mode = user._get_edi_mode()
+
+    def _get_proxy_urls(self):
+        # To extend
+        return {}
+
+    def _get_edi_mode(self):
+        edi_mode = self.env['ir.config_parameter'].sudo().get_param('account_edi_proxy_client.demo', False)
+        return 'prod' if edi_mode in ('prod', False) else 'test' if edi_mode == 'test' else 'demo'
 
-    def _get_server_url(self):
-        return DEFAULT_TEST_SERVER_URL if self._get_demo_state() == 'test' else self.env['ir.config_parameter'].sudo().get_param('account_edi_proxy_client.edi_server_url', DEFAULT_SERVER_URL)
+    def _get_server_url(self, proxy_type=None):
+        proxy_type = proxy_type or self.proxy_type
+        edi_mode = self._get_edi_mode()
+        proxy_urls = self._get_proxy_urls()
+        if not (proxy_urls.get(proxy_type) and proxy_urls[proxy_type].get(edi_mode)):
```

Comment:
Think we can let this traceback ... it's not a configuration, it's based on code. 
I'd rather have a test that ensures the configuration is correct, then a UserError at runtime.
But I'd like to hear your opinion on this :)